test/src/test/java/org/corfudb/runtime/checkpoint/CheckpointLivenessUpdaterUnitTest.java

```diff
@@ -66,4 +72,34 @@ public void testUpdateLiveness() {
         verify(txn).putRecord(Matchers.any(), Matchers.any(), captor.capture(), Matchers.any());
         Assert.assertEquals(1, captor.getValue().getSyncHeartbeat());
     }
+
+    @RepeatedTest(NUM_REPETITIONS)
+    public void testUpdateLivenessRaceCondition() {
+        ActiveCPStreamMsg activeCPStreamMsg = ActiveCPStreamMsg.newBuilder().setSyncHeartbeat(0).build();
+        when((ActiveCPStreamMsg) corfuStoreEntry.getPayload()).thenReturn(activeCPStreamMsg);
+
+        //This block helps with possibly triggering a race condition
+        long endTime = System.nanoTime() + TimeUnit.NANOSECONDS.convert(LONG_INTERVAL, TimeUnit.SECONDS);
+        boolean clear = true;
+        while (System.nanoTime() < endTime) {
```

Comment:
Is such a long delay necessary? In conjunction with the 5 repetitions, this will take over five minutes alone. Ideally, this test should be made more deterministic.

Reply:
Modified the test and CheckpointLivenessUpdater code after our offline discussion. Now the test is more deterministic and eliminates running it repeatedly.